### PR TITLE
Expose CandidateKind, add srflx/relayed candidate initializers

### PR DESCRIPTION
**Pull request by morajabi**

### CODE REVIEW

**Review by algesten — APPROVED** [review]

**Comment by algesten** [comment]
Hey @morajabi!

Maybe we should rename this PR? I think we can merge. Server reflexive looks ok to me.

**Comment by algesten** [comment]
Just a little lint fix. Looks trivial.

**Comment by algesten** [comment]
Hey @morajabi do you want to land this one?

File: ice/src/candidate.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ impl fmt::Debug for Candidate {
 
 impl Candidate {
     #[allow(clippy::too_many_arguments)]
-    fn new(
+    pub fn new(
```
Comment (algesten) [inline ice/src/candidate.rs:99]: This should not be public.

**Review by algesten — CHANGES_REQUESTED** [review]

**Comment by morajabi** [comment]
@algesten Sorry this took this long. Applied changes you requested. Please let me know what you think 🙏
(P.S. str0m has made HUGE progress, thanks to all of you, we're happily using str0m in production for months in our desktop app for P2P usecase)

File: src/ice/candidate.rs [hunk 2]
```diff
@@ -171,6 +171,48 @@ impl Candidate {
         ))
     }
 
+    /// Creates a server reflexive ICE candidate.
+    ///
+    /// Server reflexive candidates are local sockets mapped to external ip discovered via a STUN binding request.
```
Comment (algesten) [inline src/ice/candidate.rs:176]: Break this line up around ~100 ish line length, please.

File: src/ice/candidate.rs [hunk 3]
```diff
@@ -171,6 +171,48 @@ impl Candidate {
...
+            Some(addr),
+            CandidateKind::ServerReflexive,
+            None,
+            None,
+        ))
+    }
+
+    /// Creates a relayed ICE candidate.
+    ///
+    /// Relayed candidates are server sockets relaying traffic to a local socket. Allocate a TURN addr to use as a local candidate.
```
Comment (algesten) [inline src/ice/candidate.rs:197]: Ditto

**Review by algesten — CHANGES_REQUESTED** [review]
Looks good!

Just some lint error then we land it!

**Review by algesten — APPROVED** [review]

**Pull request merged**